src/dist/manifestation.rs

```diff
@@ -168,7 +169,7 @@ impl Manifestation {
 
         // Uninstall components
         for component in &update.components_to_uninstall {
-            let notification = if altered {
+            let notification = if explicit_modify {
```

Comment:
Should this not be `if !explicit_modify {` ?  It seems to me that only implicit modifications (i.e. toolchain updates) should trigger the `RemovingOldComponent` variant.

Reply:
i.e. I think you've got the boolean sense of this variable backwards -- If you call it `explicit_modify` as you have, then you need to invert this test and the true/false at the callsites.  Alternatively call it `implicit_modify` and it'll be good as-is :D